packages/components/tests/lib/options/options.test.js

```diff
@@ -1,19 +1,17 @@
 /* eslint-env jest */
 /* eslint-disable no-unused-vars */
+import { Logger } from 'alpheios-data-models'
 import Options from '@/lib/options/options'
 // import OptionItem from '@/lib/options/options-item'
 import StorageAdapter from '@/lib/options/storage-adapter'
 import LocalStorageArea from '@/lib/options/local-storage-area'
 
 describe('options.test.js', () => {
-  console.error = function () {}
-  console.log = function () {}
-  console.warn = function () {}
+  const logger = Logger.getInstance({ verbose: true })
```

Comment:
As I could see finally logger use the same console commands, 
and this stub construction should continue to work (if verbose mode is defined right)

This way I believe that it is enough in `beforeAll` block define logger inside each test with verboseMode
and tests should work well
And there is no big value in redefining the whole tracking workflow

We have this tracking construction in the most of all tests in all packages, I think
So I believe the change with defining logger should be done in each of them

Reply:
I was checking `Logger.log()` calls instead of `console.log()` to make sure it goes through our logger and does not bypass it, since the requirement is to always use our logger, not the `console`. But if we would check for the `console` calls, we would skip that check and calls to `console` functions might sneak it. Does it sound reasonable?

Reply:
For now with your code we have the following problem:
if we won't define Logger with verbose mode: true, we could fail all tests that have something like the following

`expect(console.error).toHaveBeenCalledWith(expect.stringMatching(/storing/))`
`expect(console.warn).not.toHaveBeenCalledWith(expect.stringMatching(/IndexedDB/))`

>I was checking Logger.log() calls instead of console.log() to make sure it goes through our logger and does not bypass it, since the requirement is to always use our logger

I believe that it is worth to create a separate test that would should show
that with different verbose modes we could catch console.warn and console.error in some one module
And it is not very consistent to change the overall workflow of one test file for this purpose
but of course it is from my point of view :)

Reply:
> if we won't define Logger with verbose mode: true, we could fail all tests that have something like the following

From my point of view (and I can be wrong) if any test is checking calls to `console.error()` within `Logger.error()` they should switch to checking for `Logger.error()` calls instead. The knowledge that `Logger.error()` is using `console.error()` underneath are internal implementation details. Tests cases should not rely on the knowledge of internal implementation details of the class not being tested (and, on my opinion, internal details of a class being tested should not be accessed directly too, but there are different approaches and there are exceptions).

> And it is not very consistent to change the overall workflow of one test file for this purpose

I agree that it breaks the consistency across test cases. So we might switch to testing for `console` calls everywhere for it to be consistent. It might be worth it.

There are many different approaches to testing, and each has its own benefits. It's often hard to make an objective decision about which one to use. I do not have a very strong opinion about one or the other. I think Irina's point about consistency in test suits is very good, and it might make sense to sacrifice purity for that. @balmas, what do you think? What should be our approach in this case?